lib/workflows/utils.js

```diff
@@ -68,6 +69,37 @@ const check = (config) => {
   return test[0];
 };
 
+/**
+ * Get a single workflow from all workflows available
+ *
+ * @param {object} type - a content type object
+ * @param {object} allFlows - an object of all workflows
+ * @param {object} config - global configuration
+ *
+ * @returns {string|object} error message or a single workflow object
+ */
+const workflow = (type, allFlows) => {
+  let wf;
+
+  if (type.workflow) {
+    wf = allFlows.find(flow => {
+      return flow.id === type.workflow;
+    });
+  }
+  else {
+    wf = allFlows.find(flow => {
+      return flow.id === globalConfig.workflows.default;
+    });
+  }
+
+  if (wf === undefined || typeof wf !== 'object') {
+    return globalConfig.workflows.messages.missing.replace('%type', type.name).replace('%workflow', type.workflow);
```

Comment:
Same comment I have here as I have in `content/utils`, I'd prefer to use promise-as-an-interface here and not return a string for error or object for true and have to  have the places that implement this need to check for that each time